lib/galaxy/webapps/galaxy/api/jobs.py

```diff
@@ -257,148 +266,262 @@ def common_problems(self, trans: ProvidesUserContext, id, **kwd):
+            message=payload.message,
+        )
+        return JobErrorSummary(messages=messages)
+
+    @router.get(
+        "/api/jobs/{id}/inputs",
+        name="get_inputs",
+        summary="Returns input datasets created by a job.",
+    )
+    def inputs(
+        self,
+        id: Annotated[DecodedDatabaseIdField, JobIdPathParam],
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> List[JobAssociation]:
+        job = self.service.get_job(trans=trans, job_id=id)
+        return self.service.dictify_associations(trans, job.input_datasets, job.input_library_datasets)
+
+    @router.get(
+        "/api/jobs/{id}/outputs",
+        name="get_outputs",
+        summary="Returns output datasets created by a job.",
+    )
+    def outputs(
+        self,
+        id: Annotated[DecodedDatabaseIdField, JobIdPathParam],
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> List[JobAssociation]:
+        job = self.service.get_job(trans=trans, job_id=id)
+        return self.service.dictify_associations(trans, job.output_datasets, job.output_library_datasets)
+
+    @router.get(
+        "/api/jobs/{id}/parameters_display",
+        name="resolve_parameters_display",
+        summary="Resolve parameters as a list for nested display.",
+    )
+    def parameters_display_by_job(
+        self,
+        id: Annotated[DecodedDatabaseIdField, JobIdPathParam],
+        hda_ldda: Annotated[DatasetSourceType, HdaLddaQueryParam] = DatasetSourceType.hda,
+        trans: ProvidesUserContext = DependsOnTrans,
+    ) -> JobDisplayParametersSummary:
         """
-        * GET /api/jobs/{job_id}/metrics
-        * GET /api/datasets/{dataset_id}/metrics
-            Return job metrics for specified job. Job accessibility checks are slightly
-            different than dataset checks, so both methods are available.
-
-        :type   job_id: string
-        :param  job_id: Encoded job id
-
-        :type   dataset_id: string
-        :param  dataset_id: Encoded HDA or LDDA id
-
-        :type   hda_ldda: string
-        :param  hda_ldda: hda if dataset_id is an HDA id (default), ldda if
-                          it is an ldda id.
-
-        :rtype:     list
-        :returns:   list containing job metrics
+        # TODO is this still true?
+        Resolve parameters as a list for nested display. More client logic
+        here than is ideal but it is hard to reason about tool parameter
+        types on the client relative to the server. Job accessibility checks
+        are slightly different than dataset checks, so both methods are
+        available.
+
+        This API endpoint is unstable and tied heavily to Galaxy's JS client code,
+        this endpoint will change frequently.
         """
-        job = self.__get_job(trans, **kwd)
-        return summarize_job_metrics(trans, job)
+        job = self.service.get_job(trans, job_id=id, hda_ldda=hda_ldda)
+        return summarize_job_parameters(trans, job)
```

Comment:
The legacy endpoint had this large comment block attached to it. Is it still true or can I remove them?